### PR TITLE
feat(executor): Move exit code capture to controller. See #5251

### PR DESCRIPTION
**Pull request by alexec**
See #5251

### CODE REVIEW

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -1199,6 +1199,14 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
...
 		}
 	}
+	if node.Outputs != nil && node.Outputs.ExitCode == nil {
+		for _, c := range pod.Status.ContainerStatuses {
+			if c.Name == common.MainContainerName && c.State.Terminated != nil {
+				node.Outputs.ExitCode = pointer.StringPtr(fmt.Sprint(c.State.Terminated.ExitCode))
+				updated = true
+			}
+		}
+	}
```
Comment (simster7) [inline workflow/controller/operator.go:1209]: MUCH better

Comment (alexec) [reply workflow/controller/operator.go:1209]: why have buggy code in the executor where it is hard to test when you can have that ~~buggy~~ code in the controller

Comment (sarabala1979) [reply workflow/controller/operator.go:1209]: This code should run before PodGC. I think it should be. we need to test the edge cases.

Comment (alexec) [reply workflow/controller/operator.go:1209]: This runs whenever the exit code is known (because the container is terminated) but has not been set yet. 

I'm doubling-down on this refactoring.

**Comment by alexec** [comment]
Closing. Re-opening shortly.
